nistats/second_level_model.py

```diff
@@ -0,0 +1,451 @@
+                                         ' str or array' %
+                                         (cidx, type(cond)))
+            else:
+                raise ValueError('first_level_conditions is not a list')
+
+        # check confounds
+        if confounds is not None:
+            if not isinstance(confounds, pd.DataFrame):
+                raise ValueError('confounds must be a pandas DataFrame')
+            if 'subject_id' not in confounds.columns:
+                raise ValueError('confounds DataFrame must contain column'
+                                 '"subject_id"')
+            if len(confounds.columns) < 2:
+                raise ValueError('confound should contain at least 2 columns'
+                                 'one called "subject_id" and the other with'
+                                 'a given confounder')
+
+        # check design matrix
+        if design_matrix is not None:
+            if not isinstance(design_matrix, pd.DataFrame):
+                raise ValueError('design matrix must be a pandas DataFrame')
+
+        # check condition names
+        if first_level_conditions_name is not None:
+            if not isinstance(first_level_conditions_name, (list, tuple)):
+                raise ValueError('first_level_conditions_name must be a '
+                                 'list of strings')
+            if len(first_level_conditions_name) != len(first_level_conditions):
+                raise ValueError('length of first_level_conditions_name and'
+                                 ' first_level_conditions must match')
+            for name in first_level_conditions_name:
+                if not isinstance(name, str):
+                    raise ValueError('first_level_conditions_name must be a '
+                                     'list of strings')
+
+        # Build the design matrix X and list of imgs Y for GLM fit
+        if isinstance(first_level_input, pd.DataFrame):
+            # Get only first level conditions if provided
+            if first_level_conditions is not None:
+                for condition in first_level_conditions:
+                    if condition not in first_level_input['map_name'].tolist():
+                        raise ValueError('condition %s not present in'
+                                         ' first_level_input' % condition)
+                in_cond = first_level_input.apply(lambda x: x['map_name'] in
+                                                  first_level_conditions,
+                                                  axis=1)
+                first_level_input = first_level_input[in_cond]
+            # Create design matrix if necessary
+            if design_matrix is None:
+                design = _create_second_level_design(first_level_input,
+                                                     confounds)
+            # get effect maps
+            effects_maps = first_level_input['effects_map_path'].tolist()
```

Comment:
Maybe this is not the right place for this comment, yet we should remember that for mixed-effects models, you typically want to have bit the first-level effects *and variance*.